File: Leetcode (Python, JavaScript)/19. Remove Nth Node From End of List/19. Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        curr = head
        len = 0
        while curr is not None:
            len += 1
            curr = curr.next
        curr = head
        if n==len:
            return head.next
        for i in range(len-n-1):
            curr = curr.next
        curr.next = curr.next.next
        return head

File: Leetcode (Python, JavaScript)/19. Remove Nth Node From End of List/19. Remove Nth Node From End of List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let curr = head
    let len = 0
    while (curr!=null){
        len += 1
        curr = curr.next
    }
    let curr1 = head
    if (n==len){
        return head.next
    }
    for (i=0;i<len-n-1;i++){
        curr1 = curr1.next
    }
    curr1.next = curr1.next.next
    return head
};